src/module.c

```diff
@@ -862,6 +893,86 @@ int RM_CreateCommand(RedisModuleCtx *ctx, const char *name, RedisModuleCmdFunc c
     return REDISMODULE_OK;
 }
 
+void extendKeySpecsIfNeeded(struct redisCommand *cmd) {
+    /* We extend even if keys_specs_num == keys_specs_max because
+     * this function is called prior to adding a new spec */
+    if (cmd->keys_specs_num < cmd->keys_specs_max)
+        return;
+
+    cmd->keys_specs_max++;
+
+    if (cmd->keys_specs == cmd->keys_specs_static) {
+        cmd->keys_specs = zmalloc(sizeof(keysSpec) * cmd->keys_specs_max);
+        memcpy(cmd->keys_specs, cmd->keys_specs_static, sizeof(keysSpec) * cmd->keys_specs_num);
+    } else {
+        cmd->keys_specs = zrealloc(cmd->keys_specs, sizeof(keysSpec) * cmd->keys_specs_max);
+    }
+}
+
+int moduleAddCommandKeySpec(RedisModuleCtx *ctx, const char *name, const char *specflags, keysSpec *spec) {
+    int64_t flags = specflags ? commandKeySpecsFlagsFromString((char*)specflags) : 0;
+    if (flags == -1)
+        return REDISMODULE_ERR;
+
+    struct redisCommand *cmd = lookupCommandByCString(name);
+
+    if (cmd == NULL)
+        return REDISMODULE_ERR;
+
+    if (!(cmd->flags & CMD_MODULE))
+        return REDISMODULE_ERR;
+
+    RedisModuleCommandProxy *cp = (RedisModuleCommandProxy*)(unsigned long)cmd->getkeys_proc;
+    if (cp->module != ctx->module)
+        return REDISMODULE_ERR;
+
+    extendKeySpecsIfNeeded(cmd);
+
+    cmd->keys_specs[cmd->keys_specs_num] = *spec; /* Set the spec */
+    cmd->keys_specs[cmd->keys_specs_num].flags = flags; /* Set the flags */
+    cmd->keys_specs_num++;
+
+    /* Refresh legacy range */
+    populateCommandLegacyRangeSpec(cmd);
+    return REDISMODULE_OK;
+}
+
+/* Add a "range" key arguments spec to a command.
+ * Returns REDISMODULE_OK */
```

Comment:
i think we need to improve the doc comment here and explain what's a "range".
maybe refer to the COMMAND command docs.

Reply:
@guybe7 i guess you forgot about this one.
let's just mention in the comment something like: `(to be used for COMMAND command output)`
this way people can go look the meaning of these flags in the docs.

the alternative is to document the meaning of `step`, `startfrom` etc, including the negative notations.
maybe it's not a bad idea. (i.e. do both, mention COMMAND command output, and the meaning of each arg).